hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -208,13 +209,17 @@ public HoodieWriteClient(JavaSparkContext jsc, HoodieWriteConfig clientConfig, b
      *
      * This implementation uses sortBy (which does range partitioning based on reservoir sampling) and
      * attempts to control the numbers of files with less memory compared to the {@link
-     * HoodieWriteClient#insert(JavaRDD, String)}
+     * HoodieWriteClient#insert(JavaRDD, String)}. Optionally it allows users to specify their own partitioner. If
+     * specified then it will be used for repartitioning records. See {@link com.uber.hoodie.table.BulkInsertPartitioner}.
      *
      * @param records    HoodieRecords to insert
      * @param commitTime Commit Time handle
+     * @param bulkInsertPartitioner If specified then it will be used to partition input records before they are
+     * inserted into hoodie.
      * @return JavaRDD[WriteStatus] - RDD of WriteStatus to inspect errors and counts
      */
-    public JavaRDD<WriteStatus> bulkInsert(JavaRDD<HoodieRecord<T>> records, final String commitTime) {
+    public JavaRDD<WriteStatus> bulkInsert(JavaRDD<HoodieRecord<T>> records, final String commitTime,
+        Option<BulkInsertPartitioner> bulkInsertPartitioner) {
```

Comment:
Can you just pass in a plain spark partitioner?  Seems like all that is being passed in, is the input RDD and a parallelism..